packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -445,20 +454,56 @@ const DataTable = ({
     ],
   );
 
-  const dataTableFilters = useMemo(
-    () =>
-      columns
-        .filter((column) => column.enableColumnFilter !== false)
-        .map((column) => {
-          return {
-            id: column.accessorKey as string,
-            label: column.header,
-            variant: column.filterVariant ?? "text",
-            options: column.filterSelectOptions,
-          } as DataFilter;
-        }),
-    [columns],
-  );
+  const convertColumnToFilter = (column: DataTableColumn<DataTableRowData>) => {
+    if (column.enableColumnFilter === false) {
+      return;
+    }
+
+    return {
+      id: column.accessorKey as string,
+      label: column.header,
+      variant: column.filterVariant ?? "text",
+      options: column.filterSelectOptions,
+    } as DataFilter;
+  };
+
+  const isDataTableColumn = (
+    item: unknown,
+  ): item is DataTableColumn<DataTableRowData> => {
+    return (
+      !!item &&
+      typeof (item as DataTableColumn<DataTableRowData>).accessorKey !==
+        "undefined"
+    );
+  };
```

Comment:
When casting as a boolean, I like to be explicit using the native `Boolean` function than using "not when the item is not the item":
```suggestion
    return (
      Boolean(item) &&
      typeof (item as DataTableColumn<DataTableRowData>).accessorKey !==
        "undefined"
    );
  };
```

Also, does `item` need to be set to an accessor key? Why not just set it as an optional accessor key in the arg?

```suggestion
  const isDataTableColumn = (
    item?: DataTableColumn<DataTableRowData>).accessorKey,
  ): item is DataTableColumn<DataTableRowData> => {
    return (
      Boolean(item) &&
      typeof item !==
        "undefined"
    );
  };
```

I mean, if it could be undefined, does that mean someone didn't pass it in or the value was `undefined`?

Making an arg optional is different than saying it could be undefined, so I'm wondering if you intend this instead:

```suggestion
  const isDataTableColumn = (
    item: (
      | DataTableColumn<DataTableRowData>).accessorKey
      | undefined
    ),
  ): item is DataTableColumn<DataTableRowData> => {
    return (
      Boolean(item) &&
      typeof item !==
        "undefined"
    );
  };
```